src/AspNetCoreModuleV2/AspNetCore/src/applicationinfo.cpp

```diff
@@ -218,26 +219,40 @@ APPLICATION_INFO::EnsureApplicationCreated(
         //
         if (!m_fAppOfflineFound)
         {
-
+            m_appStatus = APPLICATION_STATUS::FAIL;
             // Move the request handler check inside of the lock
             // such that only one request finds and loads it.
             // FindRequestHandlerAssembly obtains a global lock, but after releasing the lock,
             // there is a period where we could call
 
-            RETURN_IF_FAILED(FindRequestHandlerAssembly(struExeLocation));
+            FINISHED_IF_FAILED(FindRequestHandlerAssembly(struExeLocation));
```

Comment:
Don't mix return and Finished in one method.

Reply:
fixed

Reply:
There are still `return S_OK;` in this method